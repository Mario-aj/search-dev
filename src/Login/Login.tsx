import { LoginIcon } from '../ui';

const Login = () => {
  return (
    <div className="items-center justify-center w-full h-full gap-4 bg-gray-200 rounded-lg md:flex md:w-3/4 lg:w-3/5">
      <LoginIcon />
      <div className="flex flex-col items-center justify-center w-full h-full p-2 md:w-1/2">
        <div className="flex flex-col items-center justify-center mb-8">
          <h1 className="text-4xl font-bold text-gray-800 uppercase">
            Welcome
          </h1>
          <p className="text-lg text-center text-gray-500">
            Click on the button to login with your github
          </p>
        </div>
        <button
          type="button"
          className="items-center justify-center px-4 py-3 text-xl font-bold text-white uppercase transition-all duration-300 bg-indigo-500 rounded-md hover:bg-indigo-600 active:bg-indigo-700"
        >
          login with github
        </button>
      </div>
    </div>
  );
};

export default Login;
