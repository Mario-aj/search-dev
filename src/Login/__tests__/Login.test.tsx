import { render, screen } from '@testing-library/react';

import Login from '../Login';

describe('Login', () => {
  describe('Behavior', () => {
    it('should render correctly', () => {
      render(<Login />);

      expect(screen.getByTitle(/LoginIcon/i)).toBeInTheDocument();
      expect(
        screen.getByRole('heading', { name: /welcome/i })
      ).toBeInTheDocument();
      expect(
        screen.getByText(/click on the button to login with your github/i)
      ).toBeInTheDocument();

      const link = screen.getByRole('link', { name: /login with github/i });
      expect(link).toBeInTheDocument();
      expect(link.childNodes.length).toBe(2);
    });
  });

  describe('Styles', () => {
    it('matches snapshots', () => {
      render(<Login />);

      expect(screen.getByTitle(/LoginIcon/i)).toMatchSnapshot();
      expect(
        screen.getByRole('heading', { name: /welcome/i })
      ).toMatchSnapshot();
      expect(
        screen.getByText(/click on the button to login with your github/i)
      ).toMatchSnapshot();
      expect(
        screen.getByRole('link', { name: /login with github/i })
      ).toMatchSnapshot();
    });
  });
});
