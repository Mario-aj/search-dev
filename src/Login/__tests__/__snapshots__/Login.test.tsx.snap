// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Styles matches snapshots 1`] = `
<div
  class="hidden w-3/4 h-full px-10 py-10 bg-indigo-500 rounded md:block"
  title="LoginIcon"
>
  <svg
    data-name="Layer 1"
    height="auto"
    id="a87032b8-5b37-4b7e-a4d9-4dbfbe394641"
    viewBox="0 0 744.84799 747.07702"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M299.205,705.80851l-6.56-25.872a335.96693,335.96693,0,0,0-35.643-12.788l-.828,12.024-3.358-13.247c-15.021-4.29394-25.24-6.183-25.24-6.183s13.8,52.489,42.754,92.617l33.734,5.926-26.207,3.779a135.92592,135.92592,0,0,0,11.719,12.422c42.115,39.092,89.024,57.028,104.773,40.06s-5.625-62.412-47.74-101.5c-13.056-12.119-29.457-21.844-45.875-29.5Z"
      data-name="Path 1"
      fill="#f2f2f2"
      id="fa3b9e12-7275-481e-bee9-64fd9595a50d"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M361.591,677.70647l7.758-25.538a335.93951,335.93951,0,0,0-23.9-29.371l-6.924,9.865,3.972-13.076c-10.641-11.436-18.412-18.335-18.412-18.335s-15.315,52.067-11.275,101.384l25.815,22.51-24.392-10.312a135.91879,135.91879,0,0,0,3.614,16.694c15.846,55.234,46.731,94.835,68.983,88.451s27.446-56.335,11.6-111.569c-4.912-17.123-13.926-33.926-24.023-48.965Z"
      data-name="Path 2"
      fill="#f2f2f2"
      id="bde08021-c30f-4979-a9d8-cb90b72b5ca2"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M747.327,253.4445h-4.092v-112.1a64.883,64.883,0,0,0-64.883-64.883H440.845a64.883,64.883,0,0,0-64.883,64.883v615a64.883,64.883,0,0,0,64.883,64.883H678.352a64.883,64.883,0,0,0,64.882-64.883v-423.105h4.092Z"
      data-name="Path 22"
      fill="#e6e6e6"
      id="b3ac2088-de9b-4f7f-bc99-0ed9705c1a9d"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M680.97,93.3355h-31a23.02,23.02,0,0,1-21.316,31.714H492.589a23.02,23.02,0,0,1-21.314-31.714H442.319a48.454,48.454,0,0,0-48.454,48.454v614.107a48.454,48.454,0,0,0,48.454,48.454H680.97a48.454,48.454,0,0,0,48.454-48.454h0V141.7885a48.454,48.454,0,0,0-48.454-48.453Z"
      data-name="Path 23"
      fill="#fff"
      id="b2715b96-3117-487c-acc0-20904544b5b7"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M531.234,337.96451a24.437,24.437,0,0,1,12.23-21.174,24.45,24.45,0,1,0,0,42.345A24.43391,24.43391,0,0,1,531.234,337.96451Z"
      data-name="Path 6"
      fill="#ccc"
      id="b06d66ec-6c84-45dd-8c27-1263a6253192"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M561.971,337.96451a24.43594,24.43594,0,0,1,12.23-21.174,24.45,24.45,0,1,0,0,42.345A24.43391,24.43391,0,0,1,561.971,337.96451Z"
      data-name="Path 7"
      fill="#ccc"
      id="e73810fe-4cf4-40cc-8c7c-ca544ce30bd4"
      transform="translate(-227.576 -76.46149)"
    />
    <circle
      cx="364.43401"
      cy="261.50202"
      data-name="Ellipse 1"
      fill="#6c63ff"
      id="a4813fcf-056e-4514-bb8b-e6506f49341f"
      r="24.45"
    />
    <path
      d="M632.872,414.3305h-142.5a5.123,5.123,0,0,1-5.117-5.117v-142.5a5.123,5.123,0,0,1,5.117-5.117h142.5a5.123,5.123,0,0,1,5.117,5.117v142.5A5.123,5.123,0,0,1,632.872,414.3305Zm-142.5-150.686a3.073,3.073,0,0,0-3.07,3.07v142.5a3.073,3.073,0,0,0,3.07,3.07h142.5a3.073,3.073,0,0,0,3.07-3.07v-142.5a3.073,3.073,0,0,0-3.07-3.07Z"
      data-name="Path 8"
      fill="#ccc"
      id="bbe451c3-febc-41ba-8083-4c8307a2e73e"
      transform="translate(-227.576 -76.46149)"
    />
    <rect
      data-name="Rectangle 1"
      fill="#ccc"
      height="2.047"
      id="bb28937d-932f-4fdf-befe-f406e51091fe"
      width="218.552"
      x="218.56201"
      y="447.10197"
    />
    <circle
      cx="225.46401"
      cy="427.41999"
      data-name="Ellipse 2"
      fill="#6c63ff"
      id="fcef55fc-4968-45b2-93bb-1a1080c85fc7"
      r="6.902"
    />
    <rect
      data-name="Rectangle 2"
      fill="#ccc"
      height="2.047"
      id="ff33d889-4c74-4b91-85ef-b4882cc8fe76"
      width="218.552"
      x="218.56201"
      y="516.11803"
    />
    <circle
      cx="225.46401"
      cy="496.43702"
      data-name="Ellipse 3"
      fill="#6c63ff"
      id="e8fa0310-b872-4adf-aedd-0c6eda09f3b8"
      r="6.902"
    />
    <path
      d="M660.69043,671.17188H591.62207a4.50493,4.50493,0,0,1-4.5-4.5v-24.208a4.50492,4.50492,0,0,1,4.5-4.5h69.06836a4.50491,4.50491,0,0,1,4.5,4.5v24.208A4.50492,4.50492,0,0,1,660.69043,671.17188Z"
      fill="#6c63ff"
      transform="translate(-227.576 -76.46149)"
    />
    <circle
      cx="247.97799"
      cy="427.41999"
      data-name="Ellipse 7"
      fill="#6c63ff"
      id="e12ee00d-aa4a-4413-a013-11d20b7f97f7"
      r="6.902"
    />
    <circle
      cx="270.492"
      cy="427.41999"
      data-name="Ellipse 8"
      fill="#6c63ff"
      id="f58f497e-6949-45c8-be5f-eee2aa0f6586"
      r="6.902"
    />
    <circle
      cx="247.97799"
      cy="496.43702"
      data-name="Ellipse 9"
      fill="#6c63ff"
      id="b4d4939a-c6e6-4f4d-ba6c-e8b05485017d"
      r="6.902"
    />
    <circle
      cx="270.492"
      cy="496.43702"
      data-name="Ellipse 10"
      fill="#6c63ff"
      id="aff120b1-519b-4e96-ac87-836aa55663de"
      r="6.902"
    />
    <path
      d="M969.642,823.53851H251.656c-1.537,0-2.782-.546-2.782-1.218s1.245-1.219,2.782-1.219H969.642c1.536,0,2.782.546,2.782,1.219S971.178,823.53851,969.642,823.53851Z"
      data-name="Path 88"
      fill="#3f3d56"
      id="f1094013-1297-477a-ac57-08eac07c4bd5"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M792.25256,565.92292a10.09371,10.09371,0,0,1,1.41075.78731l44.8523-19.14319,1.60093-11.81526,17.92157-.10956-1.05873,27.0982-59.19987,15.65584a10.60791,10.60791,0,0,1-.44749,1.20835,10.2346,10.2346,0,1,1-5.07946-13.68169Z"
      fill="#ffb8b8"
      transform="translate(-227.576 -76.46149)"
    />
    <polygon
      fill="#ffb8b8"
      points="636.98 735.021 624.72 735.021 618.888 687.733 636.982 687.734 636.98 735.021"
    />
    <path
      d="M615.96281,731.51778h23.64387a0,0,0,0,1,0,0v14.88687a0,0,0,0,1,0,0H601.076a0,0,0,0,1,0,0v0A14.88686,14.88686,0,0,1,615.96281,731.51778Z"
      fill="#2f2e41"
    />
    <polygon
      fill="#ffb8b8"
      points="684.66 731.557 672.459 732.759 662.018 686.271 680.025 684.497 684.66 731.557"
    />
    <path
      d="M891.68576,806.12757h23.64387a0,0,0,0,1,0,0v14.88687a0,0,0,0,1,0,0H876.7989a0,0,0,0,1,0,0v0A14.88686,14.88686,0,0,1,891.68576,806.12757Z"
      fill="#2f2e41"
      transform="translate(-303.00873 15.2906) rotate(-5.62529)"
    />
    <circle
      cx="640.3925"
      cy="384.57375"
      fill="#ffb8b8"
      r="24.56103"
    />
    <path
      d="M849.55636,801.91945a4.47086,4.47086,0,0,1-4.415-3.69726c-6.34571-35.22559-27.08789-150.40528-27.584-153.59571a1.42684,1.42684,0,0,1-.01562-.22168v-8.58789a1.489,1.489,0,0,1,.27929-.87207l2.74024-3.83789a1.47845,1.47845,0,0,1,1.14355-.625c15.62207-.73242,66.78418-2.8789,69.25586.209h0c2.48242,3.10351,1.60547,12.50683,1.4043,14.36035l.00977.19336,22.98535,146.99512a4.51238,4.51238,0,0,1-3.71485,5.13476l-14.35644,2.36524a4.52127,4.52127,0,0,1-5.02539-3.09278c-4.44043-14.18847-19.3291-61.918-24.48926-80.38672a.49922.49922,0,0,0-.98047.13868c.25781,17.60546.88086,62.52343,1.0957,78.0371l.02344,1.6709a4.51811,4.51811,0,0,1-4.09277,4.53614l-13.84375,1.25781C849.83565,801.91359,849.695,801.91945,849.55636,801.91945Z"
      fill="#2f2e41"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M852.38089,495.2538c-4.28634,2.548-6.85116,7.23043-8.32276,11.9951a113.681,113.681,0,0,0-4.88444,27.15943l-1.55553,27.60021-19.25508,73.1699c16.68871,14.1207,26.31542,10.91153,48.78049-.63879s25.03222,3.85117,25.03222,3.85117l4.49236-62.25839,6.41837-68.03232a30.16418,30.16418,0,0,0-4.86143-4.67415,49.65848,49.65848,0,0,0-42.44229-8.99538Z"
      data-name="Path 99"
      fill="#ffffff"
      id="ae7af94f-88d7-4204-9f07-e3651de85c05"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M846.12661,580.70047a10.52561,10.52561,0,0,1,1.50061.70389l44.34832-22.1972.736-12.02551,18.2938-1.26127.98041,27.4126L852.7199,592.93235a10.4958,10.4958,0,1,1-6.59329-12.23188Z"
      fill="#ffb8b8"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M902.76552,508.41151c10.91151,3.85117,12.83354,45.57369,12.83354,45.57369-12.8367-7.06036-28.24139,4.49318-28.24139,4.49318s-3.20916-10.91154-7.06034-25.03223a24.52987,24.52987,0,0,1,5.13436-23.10625S891.854,504.558,902.76552,508.41151Z"
      data-name="Path 101"
      fill="#ffffff"
      id="a6768b0e-63d0-4b31-8462-9b2e0b00f0fd"
      transform="translate(-227.576 -76.46149)"
    />
    <path
      d="M889.99122,467.53052c-3.06-2.44837-7.23517,2.00173-7.23517,2.00173l-2.4484-22.03349s-15.30095,1.8329-25.0935-.61161-11.32255,8.87513-11.32255,8.87513a78.57978,78.57978,0,0,1-.30582-13.77092c.61158-5.50838,8.56838-11.01675,22.6451-14.68932S887.6518,439.543,887.6518,439.543C897.44542,444.43877,893.05121,469.97891,889.99122,467.53052Z"
      data-name="Path 102"
      fill="#2f2e41"
      id="bfd7963f-0cf8-4885-9d3a-2c00bccda2e3"
      transform="translate(-227.576 -76.46149)"
    />
  </svg>
</div>
`;

exports[`Login Styles matches snapshots 2`] = `
<h1
  class="text-4xl font-bold text-gray-800 uppercase"
>
  Welcome
</h1>
`;

exports[`Login Styles matches snapshots 3`] = `
<p
  class="text-lg text-center text-gray-500"
>
  Click on the button to login with your github
</p>
`;

exports[`Login Styles matches snapshots 4`] = `
<a
  class="flex items-center justify-center gap-2 px-4 py-3 text-xl font-bold text-white uppercase transition-all duration-300 bg-indigo-500 rounded-md hover:bg-indigo-600 active:bg-indigo-700"
  href="https://github.com/login/oauth/authorize?scope=user&client_id=1d2146a2f85409c7e6a2&redirect_uri=http://localhost:3000/"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 496 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
    />
  </svg>
  login with github
</a>
`;
